docs/getting-started/actions.md

```diff
@@ -176,64 +290,473 @@ config.add_page(name='Test Page', content=test_page())
+        Text(my_var),
+        Button('Reset @action', onclick=reset_action()),
+        Button('Reset Shortcut', onclick=my_var.reset()),
+    )
+
+config.add_page(name='Reset Variable', content=test_page())
+```
+
+In the above example, the `Reset @action` button and the `Reset Shortcut` button have the same functionality - both reset the `my_var` variable to its default value.
+
+### `dara.core.interactivity.plain_variable.Variable.sync`
 
-You can use `dara.core.interactivity.actions.DownloadContent` action to download a file. The example below demonstrates how to download generated data to a `csv` file.
+`sync` is a convenience method to update a given `Variable` with the value sent by the component it is attached to.
+
+```python
+from dara.core import action, Variable
+from dara.components import Select, Item
+
+some_variable = Variable()
+items=[Item(label='item1', value=1), Item(label='item2', value=2)],
+
+@action
+async def sync_action(ctx: action.Ctx):
+    await ctx.update(target=some_variable, value=ctx.input)
+
+# Long form
+Select(
+    items=items,
+    onchange=sync_action()
+)
+# Short form
+Select(
+    items=items,
+    onchange=some_variable.sync()
+)
+```
+
+In the above example, the `sync_action` action and the `sync` shortcut action have the same functionality - both update the `some_variable` variable with the value of the selected item in the `Select` component.
+Note that if you would like to transform the value coming from the component before updating the variable, you should use the `@action` decorator instead.
+
+:::tip
+
+This method is also available for `UrlVariable` in addition to `Variable`.
+
+:::
+
+### `dara.core.interactivity.plain_variable.Variable.update`
+
+`update` is a convenience method to update a given `Variable` with a new static value.
+
+```python
+from dara.core import action, Variable
+from dara.components import Button
+
+var = Variable(default=False)
+
+@action
+async def manual_action(ctx: action.Ctx):
+    await ctx.update(target=var, value=True)
+
+Button('set to True', onclick=manual_action())
+Button('set to True', onclick=var.update(value=True))
+```
+
+In the above example, the `manual_action` action and the `update` shortcut action have the same functionality - both update the `var` variable with the value `True`.
+The shortcut action is useful in scenarios where you want to update a variable with a static value, without the need to compute it in an `@action`-annotated function.
+
+:::tip
+
+This method is also available for `UrlVariable` and `DataVariable` in addition to `Variable`.
+
+:::
+
+### `dara.core.interactivity.plain_variable.Variable.toggle`
+
+`toggle` is a convenience method to toggle a given `Variable` between `True` and `False`.
+
+```python
+from dara.core import action, Variable
+from dara.components import Button
+
+var = Variable(default=False)
+
+@action
+async def toggle_action(ctx: action.Ctx):
+    await ctx.update(target=var, value=not ctx.input)
```

Comment:
I am confused, isn't input of a Button always None?

Reply:
Oops good catch, will come up with a better example